conans/test/functional/only_source_test.py

```diff
@@ -1,163 +1,164 @@
-        client.run("upload Hello0/1.0@lasote/stable --all", assert_error=True)
-        self.assertIn("no packages can be uploaded", client.out)
-
-    def test_reuse(self):
-        client = TestClient(default_server_user=True)
-        ref = ConanFileReference.loads("Hello0/0.1@lasote/stable")
-        client.save({"conanfile.py": GenConanfile("Hello0", "0.1")})
-        client.run("export . lasote/stable")
-        client.run("install %s --build missing" % str(ref))
-
-        self.assertTrue(os.path.exists(client.cache.package_layout(ref).builds()[0]))
-        self.assertTrue(os.path.exists(client.cache.package_layout(ref).packages()[0]))
-
-        # Upload
-        client.run("upload %s --all" % str(ref))
-
-        # Now from other "computer" install the uploaded conans with same options (nothing)
-        other_client = TestClient(servers=client.servers, users=client.users)
-        other_client.run("install %s --build missing" % str(ref))
-        self.assertFalse(os.path.exists(other_client.cache.package_layout(ref).builds()[0]))
-        self.assertTrue(os.path.exists(other_client.cache.package_layout(ref).packages()[0]))
-
-        # Now from other "computer" install the uploaded conans with same options (nothing)
-        other_client = TestClient(servers=client.servers, users=client.users)
-        other_client.run("install %s --build" % str(ref))
-        self.assertTrue(os.path.exists(other_client.cache.package_layout(ref).builds()[0]))
-        self.assertTrue(os.path.exists(other_client.cache.package_layout(ref).packages()[0]))
-
-        # Use an invalid pattern and check that its not builded from source
-        other_client = TestClient(servers=client.servers, users=client.users)
-        other_client.run("install %s --build HelloInvalid" % str(ref))
-        self.assertIn("No package matching 'HelloInvalid' pattern", other_client.out)
-        self.assertFalse(os.path.exists(other_client.cache.package_layout(ref).builds()[0]))
-        # self.assertFalse(os.path.exists(other_client.cache.package_layout(ref).packages()))
-
-        # Use another valid pattern and check that its not builded from source
-        other_client = TestClient(servers=client.servers, users=client.users)
-        other_client.run("install %s --build HelloInvalid -b Hello" % str(ref))
-        self.assertIn("No package matching 'HelloInvalid' pattern", other_client.out)
-        # self.assertFalse(os.path.exists(other_client.cache.package_layout(ref).builds()))
-        # self.assertFalse(os.path.exists(other_client.cache.package_layout(ref).packages()))
-
-        # Now even if the package is in local store, check that's rebuilded
-        other_client.run("install %s -b Hello*" % str(ref))
-        self.assertIn("Copying sources to build folder", other_client.out)
-
-        other_client.run("install %s" % str(ref))
-        self.assertNotIn("Copying sources to build folder", other_client.out)
+def test_conan_test():
```

Comment:
Lots of changes here are not related to the intention of the PR, I would prefer to have these separated in another PR

Reply:
Agree, in general, tests:

- New tests should be written in pytest
- Old tests, with very minor changes, do only those minor things in the PR
- Submit a separate PR with the pytest-modernizing of the tests, as a pure refactor (that can be done before or after)